internal/core/src/segcore/SegmentSealedImpl.cpp

```diff
@@ -119,6 +119,41 @@ SegmentSealedImpl::LoadVecIndex(const LoadIndexInfo& info) {
     set_bit(index_ready_bitset_, field_id, true);
 }
 
+void
+SegmentSealedImpl::WarmupChunkCache(const FieldId field_id) {
+    auto& field_meta = schema_->operator[](field_id);
+    AssertInfo(field_meta.is_vector(), "vector field is not vector type");
+
+    if (!get_bit(index_ready_bitset_, field_id) &&
+        !get_bit(binlog_index_bitset_, field_id)) {
+        return;
+    }
+
+    AssertInfo(vector_indexings_.is_ready(field_id),
+               "vector index is not ready");
+    auto field_indexing = vector_indexings_.get_field_indexing(field_id);
+    auto vec_index =
+        dynamic_cast<index::VectorIndex*>(field_indexing->indexing_.get());
+    AssertInfo(vec_index, "invalid vector indexing");
+
+    auto has_raw_data = vec_index->HasRawData();
+    if (has_raw_data) {  // No need to warmup.
+        return;
+    }
+
+    auto it = field_data_info_.field_infos.find(field_id.get());
+    AssertInfo(it != field_data_info_.field_infos.end(),
+               "cannot find binlog file for field: {}, seg: {}",
+               field_id.get(),
+               id_);
+    auto field_info = it->second;
+
+    auto cc = storage::ChunkCacheSingleton::GetInstance().GetChunkCache();
+    for (const auto& data_path : field_info.insert_files) {
+        auto column = cc->Read(data_path);
```

Comment:
this could be optimized:
if the chunk cache had run out of resource (here it's disk free space), we can stop warming up as it won't improve the performance any more

Reply:
we can optimize it after we implement LRU eviction.